libs/sdk-ui-pivot/src/impl/drilling/drilledRowFactory.ts

```diff
@@ -0,0 +1,86 @@
+// (C) 2021 GoodData Corporation
+import { TableDescriptor } from "../structure/tableDescriptor";
+import { DataValue } from "@gooddata/sdk-backend-spi";
+import { getMappingHeaderUri } from "@gooddata/sdk-ui";
+import { invariant } from "ts-invariant";
+import { IGridRow } from "../data/resultTypes";
+
+/**
+ * @deprecated this is linked to deprecated API
+ */
+function extractIdsFromAttributeElementUri(uri: string): (string | null)[] {
+    const [, attributeId, , attributeValueId = null] = uri.match(/obj\/([^/]*)(\/elements\?id=)?(.*)?$/)!;
```

Comment:
Nitpick: the empty item in the middle of the array is quite error prone. We can fix this by using a non-capturing group in the regex

```ts
const previous = /obj\/([^/]*)(\/elements\?id=)?(.*)?$/
const improved = /obj\/([^/]*)(?:\/elements\?id=)?(.*)?$/
```

Then you would be able to remove the "empty" item in the destructuring (and _marginally_ improve performance).

Reply:
this function is move & rename of existing function we had in agGridUtils.ts/getIdsFromUri ... I just took what we have there and hidden it here without studying. if you really insist i will change, however I would prefer not to touch these regexp hieroglyphs :)

Reply:
I understand, I do not insist, I just like regexes, I guess :D